lens/private/compound/join-hash.scrbl

```diff
@@ -3,7 +3,8 @@
 @(require "../doc-util/main.rkt")
 
 
-@defproc[(lens-join/hash [key any/c] [lens lens?] ... ...) lens?]{
+@defproc[(lens-join/hash [key key/c] [lens (lens/c target/c value/c)] ... ...)
+         (lens/c target/c (hash/c key/c value/c #:immutable #t))]{
```

Comment:
I think instead of `key/c` and `value/c` we should just stick to `any/c` and rely on the prose and examples to make that relationship clear. A dependent contract would be the "proper" way to specify this, but that would be very verbose in the doc contract.